coderd/database/dbauthz/dbauthz.go

```diff
@@ -1469,6 +1476,13 @@ func (q *querier) GetFileTemplates(ctx context.Context, fileID uuid.UUID) ([]dat
 	return q.db.GetFileTemplates(ctx, fileID)
 }
 
+func (q *querier) GetFrobulators(ctx context.Context, arg database.GetFrobulatorsParams) ([]database.Frobulator, error) {
+	if err := q.authorizeContext(ctx, policy.ActionRead, rbac.ResourceFrobulator.WithOwner(arg.UserID.String()).InOrg(arg.OrgID)); err != nil {
+		return nil, err
+	}
+	return q.db.GetFrobulators(ctx, arg)
```

Comment:
Your method is more efficient and pretty much correct, but it does not call `auth()` on each object, which could be an issue if things changed in the future.

The correct way to do lists is `fetchWithPostFilter`, and if that is too inefficient, you can create a SQL filter. For this PR, let's switch this to `fetchWithPostFilter`.


```suggestion
return fetchWithPostFilter(q.auth, policy.ActionRead, q.db.GetFrobulators)(ctx, arg)
```